tests/bsim/bluetooth/host/adv/chain/src/main.c

```diff
@@ -76,20 +81,21 @@ static bool data_cb(struct bt_data *data, void *user_data)
 static void scan_recv(const struct bt_le_scan_recv_info *info,
 		      struct net_buf_simple *buf)
 {
-	static uint8_t sid[CONFIG_BT_EXT_ADV_MAX_ADV_SET];
+	static uint8_t sid[MAX_EXT_ADV_SET];
```

Comment:
If `#define MAX_EXT_ADV_SET (CONFIG_BT_EXT_ADV_MAX_ADV_SET - 1)`, then won't we risk `sid[0]` if this is built with `CONFIG_BT_EXT_ADV_MAX_ADV_SET=1`?

If we never want this test to support that config value, suggest to add a BUILD_ASSERT with an explanation